core/src/host.rs

```diff
@@ -348,7 +349,19 @@ impl HostEnv {
             deployed_contracts: RefCell::new(vec![]).into(),
             events_count: Rc::new(RefCell::new(Default::default())),
             native_events_count: Rc::new(RefCell::new(Default::default())),
-            events_initialized: Rc::new(RefCell::new(Default::default()))
+            events_initialized: Rc::new(RefCell::new(Default::default())),
+            captures_events: Rc::new(RefCell::new(true))
+        }
+    }
+
+    /// Sets the `captures_events` flag, which determines whether events should be captured.
+    pub fn set_captures_events(&self, captures: bool) {
+        *self.captures_events.borrow_mut() = captures;
+        if captures {
```

Comment:
I'd only do this if capturing was `false`. Otherwise if called twice with `captures == true` it would refresh contracts twice for no reason.

Reply:
It only initializes, so if it's been init already then nothing happens, except a few (depending on the number of contracts) `is init checks`.